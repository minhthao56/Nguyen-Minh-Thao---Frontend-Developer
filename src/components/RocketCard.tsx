interface RocketCardProps {
  image?: string;
  description?: string;
  name?: string;
  country?: string;
}

export default function RocketCard({
  image,
  description,
  name,
  country,
}: RocketCardProps) {
  return (
    <div className="max-w-sm rounded overflow-hidden shadow-lg">
      <img className="w-full  h-72" src={image} alt="Sunset in the mountains" />
      <div className="px-6 py-4">
        <div className="font-bold text-xl mb-2">{name}</div>
        <p className="text-gray-700 text-base line-clamp-2">{description}</p>
      </div>
      <div className="px-6 pb-2">
        <span className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2">
          {country}
        </span>
      </div>
    </div>
  );
}
