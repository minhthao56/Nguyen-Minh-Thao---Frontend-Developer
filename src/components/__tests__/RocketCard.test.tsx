import { render, screen } from "@testing-library/react";
import RocketCard, { RocketCardProps } from "../RocketCard";

describe("<RocketCard/>", () => {
  const renderComponent = (props?: Partial<RocketCardProps>) => {
    return render(<RocketCard {...props} />);
  };

  it("should match to snapshot", () => {
    const { container } = renderComponent({
      country: "USA",
      description: "description",
      image: "NO",
      name: "NAME",
    });
    expect(container).toMatchSnapshot();

    expect(screen.getByText("USA")).toBeInTheDocument();
    expect(screen.getByText("description")).toBeInTheDocument();
    expect(screen.getByText("NAME")).toBeInTheDocument();
  });
});
