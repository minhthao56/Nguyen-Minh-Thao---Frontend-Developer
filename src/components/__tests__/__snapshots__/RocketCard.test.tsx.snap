// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RocketCard/> should match to snapshot 1`] = `
<div>
  <div
    class="max-w-sm rounded overflow-hidden shadow-lg"
  >
    <img
      alt="Sunset in the mountains"
      class="w-full  h-72"
      src="NO"
    />
    <div
      class="px-6 py-4"
    >
      <div
        class="font-bold text-xl mb-2"
      >
        NAME
      </div>
      <p
        class="text-gray-700 text-base line-clamp-2"
      >
        description
      </p>
    </div>
    <div
      class="px-6 pb-2"
    >
      <span
        class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
      >
        USA
      </span>
    </div>
  </div>
</div>
`;
